research/picai/nnunet_scripts/nnunet_launch.slrm

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+#SBATCH --gres=gpu:a40:1
+#SBATCH --qos=normal
+#SBATCH --mem=64GB
+#SBATCH -c 16
+#SBATCH --time=12:00:00
+#SBATCH --signal=B:USR1@60
+#SBATCH --job-name=nnunet_picai_train
+#SBATCH --output=%j_%x.out
+#SBATCH --error=%j_%x.err
+
+# Process Inputs
+DATASET_NAME=$1
+UNET_CONFIG=$2
+FOLD=$3
+VENV_PATH=$4
+PLANS_IDENTIFIER=${5:-'plans'} # Default value of plans string
+PRETAINED_WEIGHTS=${6:-''} # Default value of empty string
+
+# Log input information
+echo "Dataset Name : ${DATASET_NAME}"
+echo "Config: ${UNET_CONFIG}"
+echo "Dataset Fold: ${FOLD}"
+echo "Python Env Path: ${VENV_PATH}"
+echo "Plans Identifier: ${PLANS_IDENTIFIER}"
+
+if [[ -z "${PRETRAINED_WEIGHTS}" ]]; then
+	echo "Pretrained Weights: ${PRETRAINED_WEIGHTS}"
+fi
+
+# Setup before running training
+source ${VENV_PATH}bin/activate
+BASE_DIR="$nnUNet_results/${DATASET_NAME}/nnUNetTrainer__${PLANS_IDENTIFIER}__${UNET_CONFIG}/"
+
+# Function returning array of args passed to nnUNetv2_train command
+get_nnunet_train_args() {
+	local checkpoint_bool=$1
```

Comment:
It looks like this is tied to the DATASET_NAME input variable? I assume it is supposed to be but I'm not quite certain why. Maybe a comment would be helpful here?

Reply:
When used within the context of a function the $1 refers to the 1st argument passed to the function, not the 1st argument of the script

Reply:
Ah I see! That makes sense. Bash is so weird

Reply:
Yeah lol so weird, I keep learning more I didnt know hahah